include/dataformats/GeoID.hpp

```diff
@@ -10,71 +10,87 @@
 #define DATAFORMATS_INCLUDE_DATAFORMATS_GEOID_HPP_
 
 #include <cstdint>
+#include <istream>
 #include <limits>
 #include <ostream>
-#include <istream>
 #include <tuple>
 
 namespace dunedaq {
 namespace dataformats {
 
+enum class GeoIDComponentType : uint16_t
+{
+  kTPC = 1,
+  kPDS = 2,
+  kDataSelection = 3,
+  kInvalid = 0
+};
+
 /**
  * @brief Represents a coordinate point in the DAQ's logical coordinate system (i.e. not physical coordinates)
  */
 struct GeoID
 {
   /**
-   * @brief An invalid APA number, used for initialization
+   * @brief An invalid region number, used for initialization
    */
-  static constexpr uint32_t s_invalid_apa_number = std::numeric_limits<uint32_t>::max(); // NOLINT(build/unsigned)
+  static constexpr uint16_t s_invalid_region_id = std::numeric_limits<uint16_t>::max(); // NOLINT(build/unsigned)
   /**
-   * @brief An invalid link number, used for initialization
+   * @brief An invalid element number, used for initialization
    */
-  static constexpr uint32_t s_invalid_link_number = std::numeric_limits<uint32_t>::max(); // NOLINT(build/unsigned)
+  static constexpr uint32_t s_invalid_element_id = std::numeric_limits<uint32_t>::max(); // NOLINT(build/unsigned)
+
+  /**
+   * @brief The type of the component (i.e. which subsystem it belongs to)
+  */
+  GeoIDComponentType component_type{ GeoIDComponentType::kInvalid };
 
   /**
-   * @brief APA Number of the component
+   * @brief Region number of the component
    */
-  uint32_t apa_number{ s_invalid_apa_number }; // NOLINT(build/unsigned)
+  uint16_t region_id{ s_invalid_region_id }; // NOLINT(build/unsigned)
   /**
-   * @brief Link Number of the component
+   * @brief Number of the component within the region
    */
-  uint32_t link_number{ s_invalid_link_number }; // NOLINT(build/unsigned)
-
+  uint32_t element_id{ s_invalid_element_id }; // NOLINT(build/unsigned)
 
   /**
    * @brief Comparison operator (to allow GeoID to be used in std::map)
    * @param other GeoID to compare
-   * @return The result of std::tuple compare using apa_number and link_number
+   * @return The result of std::tuple compare using GeoID fields
    */
   bool operator<(const GeoID& other) const
```

Comment:
Since we are already tempering with this file, can we make the comparisons operator `noexcept` ?